packages/client/hmi-client/src/services/calibrate-workflow.ts

```diff
@@ -116,27 +117,67 @@ export const renderLossGraph = (
 	yAxisGroup.attr('transform', `translate(${marginLeft}, 0)`).call(yAxis);
 };
 
-export const autoCalibrationMapping = async (modelOptions: State[], datasetOptions: any[]) => {
+export const autoCalibrationMapping = async (
+	modelOptions: State[],
+	datasetOptions: DatasetColumn[]
+) => {
 	const result = [] as CalibrateMap[];
-	modelOptions.forEach((modelOption) => {
-		datasetOptions.forEach((datasetOption) => {
-			// Check for direct string match
-			if (modelOption.id.toLowerCase() === datasetOption.name.toLowerCase()) {
-				result.push({ modelVariable: modelOption.id, datasetVariable: datasetOption.name });
-			}
-			// No direct string match, check grounding keys (if they exist)
-			else if (
-				modelOption.grounding?.identifiers &&
-				datasetOption?.metadata?.groundings?.identifiers
-			) {
-				const datasetKeys = Object.keys(datasetOption.metadata.groundings.identifiers);
-				Object.keys(modelOption.grounding.identifiers).forEach((modelGrounding) => {
-					if (datasetKeys.includes(modelGrounding)) {
-						result.push({ modelVariable: modelOption.id, datasetVariable: datasetOption.name });
-					}
-				}); // End for each grounding key
+	const allModelGroundings: string[] = [];
+	const allDataGroundings: string[] = [];
+	const acceptableDistance = 0.7;
+	// Get all model groundings
+	modelOptions.forEach((state) => {
+		if (state.grounding?.identifiers) {
+			const modelTemp = Object.entries(state.grounding?.identifiers);
+			const modelGroundingList = modelTemp.map((ele) => ele.join(':'));
+			modelGroundingList.forEach((ele) => allModelGroundings.push(ele));
```

Comment:
Why not chaining those calls?

```suggestion
		Object.entries(state.grounding?.identifiers)
		    .map((ele) => ele.join(':'))
		    .forEach(allModelGroundings.push);
```

Reply:
Good call
Will do